src/python/gudhi/simplex_tree.pyx

```diff
@@ -639,3 +639,10 @@ cdef class SimplexTree:
             self.thisptr = <intptr_t>(ptr.collapse_edges(nb_iter))
             # Delete old pointer
             del ptr
+
+    def __eq__(self, other:SimplexTree):
+        """Simplex tree equality operator using C++ depth first search operator==
```

Comment:
Is it really relevant to say it uses C++ and DFS? All that matters is that it is a structural equality.

Reply:
```suggestion
        """Test for structural equality
```